### ISSUE TITLE
To systray

### ISSUE THREAD

**Issue by LionelR** [pbek/QOwnNotes#172]
Hi,
Just use QOwnNotes for a couple of hours but I think it's going to be my main note&task program. Very great app!
For me, it just misses a systray app with the possible remaining tasks, and it will be perfect.
Cheers

**Comment by pbek** [comment]
Thank you for your kind words! There is a checkbox in the `Note` menu to allow showing QOwnNotes in the system tray. Can you please explain "remaining tasks"?

**Comment by LionelR** [comment]
I didn't see the checkbox, sorry for that. By remaining tasks, I mean not closed tasks. 

Le 18 mai 2016 20:52:27 GMT+02:00, Patrizio Bekerle <notifications@github.com> a écrit :
>Thank you for your kind words! There is a checkbox in the `Note` menu
>to allow showing QOwnNotes in the system tray. Can you please explain
>"remaining tasks"?
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/pbek/QOwnNotes/issues/172#issuecomment-220123171

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by pbek** [comment]
I'm sorry, I think that feature doesn't make much sense at the moment, tasks are not checked in the background in QOwnNotes.

**Comment by piyo-gh** [comment]
Well, I think current **Show in system tray** feature is not so useful on Windows.
When I minimized QOwnNotes window, both taskbar icon and system tray icon remain.
When I close QOwnNotes window, QOwnNotes quits and both taskbar icon and system tray icon disappear.

As optional feature, **Close to system tray** and/or **Minimize to system tray** feature is useful on Windows.

**Comment by LionelR** [comment]
So, there's no warning on start/due dates for open tasks ? 

Le 18 mai 2016 22:32:45 GMT+02:00, Patrizio Bekerle <notifications@github.com> a écrit :
>I'm sorry, I think that feature doesn't make much sense at the moment,
>tasks are not checked in the background in QOwnNotes.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/pbek/QOwnNotes/issues/172#issuecomment-220149191

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by pbek** [comment]
Only rudimentary. You will get a message box if the task was previously loaded and QOwnNotes is open at the time the reminder date is set.

**Comment by LionelR** [comment]
Ok. Is it possible to add it to a possible enhancement? It will be very great to:
- Have an option to "Close to system tray" because there's no need to display both in the taskbar and the systray, and have the systray without other information
- In the systray, on a click on the icon to display some shortcuts like "create a note", listing the 10 last notes, ..., and the opened tasks, which will be loaded in the background

**Comment by pbek** [comment]
I will look into that...

**Comment by pbek** [comment]
> Have an option to "Close to system tray" because there's no need to display both in the taskbar and the systray, and have the systray without other information

if you close the main window it will automatically be "closed to the system tray" if system tray is enabled for QOwnNotes

> In the systray, on a click on the icon to display some shortcuts like "create a note", listing the 10 last notes, ..., and the opened tasks, which will be loaded in the background

I implemented that all for v16.05.17
from the changelog:

- from the QOwnNotes tray icon you can now create new notes, open the ten top 
  most notes, open the todo lists, open the 10 most recent tasks or quit the app
  (for [Issue #172](https://github.com/pbek/QOwnNotes/issues/172))
- the tasks of your todo lists will now automatically be updated from your
  ownCloud server every 10 minutes to allow better reminding of tasks
  (for [Issue #172](https://github.com/pbek/QOwnNotes/issues/172))

can you please test the new features when I release them today?

**Comment by LionelR** [comment]
Sure I will try :)

**Comment by pbek** [comment]
I just released v16.05.17.

**Comment by LionelR** [comment]
Hum, there are some problems :
If I start QOwnNotes with the "Show in systray" previously enabled, 
clicking on the systray icon shows nothing. I have to go to the Window 
menu, disable and reenable it and after that, I have a menu in the 
systray. But, the menu is repeating himself multiple time :



And closing the app with the close window button close the systray to.

**Comment by Maboroshy** [comment]
All good on Linux.

**Comment by LionelR** [comment]
I on Ubuntu 14.04.4

**Comment by Maboroshy** [comment]
Manajaro Plasma 5 for me.

**Comment by pbek** [comment]
thank you, @Maboroshy 

@LionelR, do you use Unity? Too bad I have no 14.04 box around any more...
Are multiple instances of QOwnNotes open (can you please check in a task manager or with `ps ax | grep QOwnNotes`? Do the menu items also multiply if you reload your note folder or leave the todo list dialog?

**Comment by LionelR** [comment]
Yes I use Unity.
There's only one instance. 
No diff when reloading the note folder, but clicking on systray actions increase the number of elements

**Comment by pbek** [comment]
Thank you for reporting, I will see where I can get a Unity Desktop for testing...

**Comment by pbek** [comment]
@LionelR, can you please post the first variable block from the Debug settings.
The problem didn't exist on Ubuntu 15.10 I had and I didn't get a 14.04 together...
I now attempted to make a blind fix for QOwnNotes 16.05.18. Can you please test that when it's released and report back?

**Comment by LionelR** [comment]
Here's the debug from 16.05.18
QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `sam. mai 21 17:59:31 2016`
**Version**: `16.05.18`
**Build date**: `May 21 2016`
**Build number**: `1648`
**Platform**: `linux`
**Operating System**: `Ubuntu 14.04.4 LTS`
**Build architecture**: `x86_64`
**Current architecture**: `x86_64`
**Release**: `Travis CI`
**Qt Version**: `5.4.2`

The systray now doesn't show multiple entries, but as previously the systray is unselectable is I open QOwnNotes with systray pre-enabled. After disable/enable, the systray can be accessed, but all entries are empty. Worst, selecting some entries in the global menu cause a core dump and the app close.

**Comment by LionelR** [comment]
In fact, the app closes itself whenever I use it or not. I have removed all my local settings files, but do nothing :(

**Comment by pbek** [comment]
thank you for posting. do you also use tasks? if yes, can you access them with QOwnNotes?
and you don't see any recent notes and tasks in the tray context menu?
and if you click an other tray context menu item the app crashes?

**Comment by LionelR** [comment]
Yes I use tasks and I can access them with QOwnNotes. 
No tasks or notes in the systray. 
The app crashes by itself, if I click in the menu or do nothing

Le 21 mai 2016 23:12:36 GMT+02:00, Patrizio Bekerle <notifications@github.com> a écrit :
>thank you for posting. do you also use tasks? if yes, can you access
>them with QOwnNotes?
>and you don't see any recent notes and tasks in the tray context menu?
>and if you click an other tray context menu item the app crashes?
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pbek/QOwnNotes/issues/172#issuecomment-220800581

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by pbek** [comment]
And when does the the app crash by itself? After how much run-time or when what happens? Which version of QOwnNotes was the first where that happened?

**Comment by pbek** [comment]
I finally got a Ubuntu 14.04 with Qt 5.5.1 and Qt Creator running to debug. I made a hot-fix in QOwnNotes 15.06.19 (just released) for the crash, but the tray icon still doesn't work and I found no solution to it yet. It just doesn't work like it is supposed to according to the Qt documentation... Maybe that's a Unity thing...

**Comment by LionelR** [comment]
The app crashes a few seconds after startup if I have activated a calendar in the tasks settings.
The app crashes each time the settings dialog box is closed, on a click on the cancel/OK buttons, or on the close window manager button. This also happens with the tasks window.
First time : 16.05.17

**Comment by pbek** [comment]
Please try QOwnNotes 15.06.19 if it fixes at least the crash.

**Comment by LionelR** [comment]
Yes, I saw your last post just after mine :)

**Comment by pbek** [comment]
Great. ;) But I'm not sure yet if the system tray menu problem is solvable under Ubuntu 14.04. It seems to work fine on my Ubuntu 15.10 test system...

**Comment by LionelR** [comment]
Ok, the crash seems to be gone.
For the systray, the multiple entries come back

**Comment by pbek** [comment]
Yes, that's an Ubuntu 14.04 thing... I've so far no clue how to fix it... Suggestions are welcome.

**Comment by pbek** [comment]
It's ironic. You had a feature request, I implemented it (was quite some work) and everyone except you can use it. :)

**Comment by LionelR** [comment]
:-)

**Comment by pbek** [comment]
I invested some more hours, but still found no workaround for it... I even found some other issues with the systray that vary from Linux distribution to Linux distributions (for example if you disable the systray and enable it again the icon doesn't work any more)... There is no workaround for them too...
It seems to be an issue of Qt with different desktop environments...

I will close this issue for now, maybe I stumble upon a solution for it some day.
